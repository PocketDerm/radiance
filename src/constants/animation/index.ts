export default {
  defaultTiming: '350ms',
};
