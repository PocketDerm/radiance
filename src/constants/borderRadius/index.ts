const BORDER_RADIUS = {
  small: '4px',
  medium: '8px',
  large: '32px',
} as const;

export default BORDER_RADIUS;
