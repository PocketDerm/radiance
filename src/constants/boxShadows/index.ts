/*
 * Note: Each Category is exported separately to be rendered in storybook:
 */

// @ts-ignore
import { transparentize } from '../../utils';
import COLORS from '../colors';

const boxShadowColor = `${COLORS.primary}`;
const boxShadowOverlayColor = '#505050';

export const BASE_CONFIG = {
  clickable: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.06)}`,
  clickableHover: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  message: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  focusSecondary: `0 0 2px 2px ${transparentize(COLORS.secondary, 0.5)}`,
};

export const OLD_BASE_CONFIG = {
  boxShadow1: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  boxShadow2: `0 5px 11px 0 ${transparentize(boxShadowColor, 0.1)}`,
  boxShadow3: `0 4px 6px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow4: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow5: `0 1px 2px 0 ${transparentize(boxShadowColor, 0.06)}`,
  boxShadowOnOverlay: `0 12px 20px 0 ${boxShadowOverlayColor}`,
};

export const DESIGN_SYSTEM_NAMING = {
  elevation2: OLD_BASE_CONFIG.boxShadow5,
  elevation3: OLD_BASE_CONFIG.boxShadow4,
  elevation6: OLD_BASE_CONFIG.boxShadow3,
  elevation11: OLD_BASE_CONFIG.boxShadow2,
  elevation20: OLD_BASE_CONFIG.boxShadow1,
};

export const ALIASED_CONFIG = {
  card: OLD_BASE_CONFIG.boxShadow4,
};

export default {
  ...BASE_CONFIG,
  ...OLD_BASE_CONFIG,
  ...DESIGN_SYSTEM_NAMING,
  ...ALIASED_CONFIG,
};
