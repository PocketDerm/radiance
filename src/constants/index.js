export { default as ANIMATION } from './animation';
export { default as BOX_SHADOWS } from './boxShadows';
export { default as BREAKPOINTS } from './breakpoints';
export { default as COLORS } from './colors';
export { default as FONTS } from './fonts';
export { default as GRID } from './grid';
export { default as LAYOUT } from './layout';
export { default as MEDIA_QUERIES } from './mediaQueries';
export { default as PROGRESS_BAR_STATUS } from './progressBarStatus';
export { default as SPACING } from './spacing';
export { default as SPACER } from './spacer';
export { default as TYPOGRAPHY_CONSTANTS } from './typography';
export { default as Z_SCALE } from './zScale';
