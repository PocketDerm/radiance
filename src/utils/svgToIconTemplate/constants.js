const UTIL_LOCATION = '../../../utils/icons';

module.exports = {
  UTIL_LOCATION,
};
