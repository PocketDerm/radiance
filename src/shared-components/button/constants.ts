import { COLORS } from '../../constants';

// TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
  [COLORS.primary]: {
    tint1: COLORS.primaryTint1,
    tint2: COLORS.primaryTint2,
  },
  [COLORS.secondary]: {
    tint1: COLORS.lavender80,
    tint2: COLORS.lavender60,
  },
};
