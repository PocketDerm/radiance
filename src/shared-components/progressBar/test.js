import React from 'react';
import renderer from 'react-test-renderer';

import { PROGRESS_BAR_STATUS } from '../../constants';

import ProgressBar from './index';

describe('<ProgressBar />', () => {
  describe('UI snapshot', () => {
    it('renders content and children', () => {
      const component = renderer.create(
        <ProgressBar status={PROGRESS_BAR_STATUS.loading} />,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
