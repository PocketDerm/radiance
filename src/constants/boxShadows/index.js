import { transparentize } from 'src/utils';

const boxShadowColor = '#2d2d30';
const boxShadowOverlayColor = '#505050';

const BASE_CONFIG = {
  clickable: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.06)}`,
  clickableHover: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  message: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
};

const OLD_BASE_CONFIG = {
  boxShadow1: `0 12px 20px 0 ${transparentize(boxShadowColor, 0.05)}`,
  boxShadow2: `0 5px 11px 0 ${transparentize(boxShadowColor, 0.1)}`,
  boxShadow3: `0 4px 6px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow4: `0 1px 3px 0 ${transparentize(boxShadowColor, 0.15)}`,
  boxShadow5: `0 1px 2px 0 ${transparentize(boxShadowColor, 0.06)}`,
  boxShadowOnOverlay: `0 12px 20px 0 ${boxShadowOverlayColor}`,
};

const DESIGN_SYSTEM_NAMING = {
  elevation2: OLD_BASE_CONFIG.boxShadow5,
  elevation3: OLD_BASE_CONFIG.boxShadow4,
  elevation6: OLD_BASE_CONFIG.boxShadow3,
  elevation11: OLD_BASE_CONFIG.boxShadow2,
  elevation20: OLD_BASE_CONFIG.boxShadow1,
};

const ALIASED_CONFIG = {
  card: OLD_BASE_CONFIG.boxShadow4,
};

export default {
  ...BASE_CONFIG,
  ...OLD_BASE_CONFIG,
  ...DESIGN_SYSTEM_NAMING,
  ...ALIASED_CONFIG,
};

