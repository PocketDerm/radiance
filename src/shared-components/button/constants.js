import { COLORS } from 'src/constants';

//TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = {
    purple: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    primary: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    purple100: {
    	tint1: COLORS.purple85,
    	tint2: COLORS.purple70,
    },
    lavender100: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
    secondary: {
    	tint1: COLORS.lavender80,
    	tint2: COLORS.lavender60,
    },
};
