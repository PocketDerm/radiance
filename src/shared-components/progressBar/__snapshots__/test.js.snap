// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> UI snapshot renders content and children 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
  }

  to {
    -webkit-transform: translateX(-5%);
    -ms-transform: translateX(-5%);
    transform: translateX(-5%);
  }
}

.emotion-2 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  background: #f8f8fa;
  overflow: hidden;
  -webkit-transition: opacity ease-in-out 500ms;
  transition: opacity ease-in-out 500ms;
  width: 100%;
  z-index: 2;
  height: 4px;
}

.emotion-0 {
  position: absolute;
  width: 100%;
  height: 4px;
  background-color: #332e54;
  -webkit-animation: animation-0 20s ease-in;
  animation: animation-0 20s ease-in;
  -webkit-transform: translateX(-5%);
  -ms-transform: translateX(-5%);
  transform: translateX(-5%);
  -webkit-transition: -webkit-transform ease-in-out;
  -webkit-transition: transform ease-in-out;
  transition: transform ease-in-out;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform ease-in-out;
  -webkit-transition: transform ease-in-out;
  transition: transform ease-in-out;
}

<div
  className="emotion-2 emotion-3"
  height={4}
>
  <div
    className="emotion-0 emotion-1"
    height={4}
  />
</div>
`;
